Automattic/jetpack
# Post editor: Error when trying to use an image from Pexels

Fixes:
- https://github.com/Automattic/wp-calypso/issues/97213

## Proposed changes:

* We check if the media is defined when performing the API call.
* If the media isn't defined we trigger an error instead of continuing with the flow and crashing the editor.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Apply the code on your sandbox
* Sandbox your API
* Apply this to your Jetpack code (the patch will not work if applied directly, change manually), with the patch applied you should be able to reproduce the original error reported on the parent issue:
```
diff --git a/wp-content/mu-plugins/jetpack-plugin/sun/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php b/wp-content/mu-plugins/jetpack-plugin/sun/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php
index e4a4aea0a8..b6f8f295cb 100644
--- _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php
+++ _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php
@@ -337,6 +337,12 @@ class WPCOM_REST_API_V2_Endpoint_External_Media extends WP_REST_Controller {
         * @return array|\WP_Error|mixed
         */
        public function get_external_media( \WP_REST_Request $request ) {
+               return [
+                       'meta'  => [
+                               'account' => 'any_account',
+                       ],
+               ]; 
+
                $params     = $request->get_params();
                $wpcom_path = sprintf( '/meta/external-media/%s', rawurlencode( $params['service'] ) );
```
* Go to the post editor and select a featured image using Pexels
* You should see an error instead of the editor crashing.

https://github.com/user-attachments/assets/77b16b3a-0cb1-4f3d-ab32-2fdb23bf647f
